### DISCUSSION TITLE
Content Search Issue in Home Tab

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#709] [General Discussion]
Hello Team,

We are on release 5.0.
When attempting to search for content using either (do_id) or name in the Home tab of our dev environment, the screen continuously loads without displaying any results. This issue is specific to the dev environment and is not occurring in the production environment.

Observations:
Upon investigation, it has been found that in the dev environment, the read API for the following request is not being called. But when i am reading the same form through Postman, I am getting a status of 200, which means the form is configured. 
`{
    "type": "schemas",
    "action": "get",
    "subType": "search",
    "rootOrgId": "*"
}
`

Actions Taken:
1. I attempted to perform content searches using various parameters in the Home tab.
2. I reviewed the network and observed that the read API mentioned above is not triggered in the dev environment.
3. I compared configurations between dev and prod environments.

Screenshots:
This is the dev env where we are facing issues:
![image](https://github.com/Sunbird-Ed/Community/assets/72926048/c3f8d729-6a3e-488a-808d-51e16c664db0)

This is the prod env where the search is working:
![image](https://github.com/Sunbird-Ed/Community/assets/72926048/02fd77f8-aa62-418f-8ade-7547cc7fd459)

Please provide any additional guidance or assistance to resolve this issue. Thank you.

**Comment by rajeevsathish** [comment]
@siddhishinde0723 can you please pass me the url we can see what is the form that is failing 

what I can see is this form is not there in dev please check if this is set properly
`curl --location -g --request POST '{{host}}/api/data/v1/form/read' \
--header 'Accept: application/json' \
--header 'Content-Type: application/json' \
--data-raw '{
    "request": {
        "type": "schemas",
        "action": "get",
        "subType": "search",
        "rootOrgId": "*",
        "framework": "*"
    }
}'`

**Reply by siddhishinde0723** [reply]
@rajeevsathish  The url for dev is [link] (https://dev.sunbirdsaas.com/)
Also, i have checked the form, which you mention this is the response for the same 
`{
    "id": "api.form.read",
    "params": {
        "resmsgid": "0a4cb871-b871-4832-800c-affecc5f23ac",
        "msgid": "7fe0c65f-4ec4-4cf8-95ed-3234e42c55eb",
        "status": "successful"
    },
    "responseCode": "OK",
    "result": {
        "form": {
            "type": "schemas",
            "subtype": "search",
            "action": "get",
            "component": "*",
            "framework": "*",
            "data": {
                "templateName": "defaultTemplate",
                "action": "get",
                "fields": [
                    {
                        "id": "content",
                        "schema": {
                            "properties": [
                                "name",
                                "code",
                                "createdOn",
                                "lastUpdatedOn",
                                "status",
                                "channel",
                                "mimeType",
                                "osId",
                                "contentEncoding",
                                "contentDisposition",
                                "mediaType",
                                "os",
                                "minOsVersion",
                                "compatibilityLevel",
                                "minGenieVersion",
                                "minSupportedVersion",
                                "filter",
                                "variants",
                                "config",
                                "visibility",
                                "audience",
                                "posterImage",
                                "badgeAssertions",
                                "targets",
                                "contentCredits",
                                "appIcon",
                                "grayScaleAppIcon",
                                "thumbnail",
                                "screenshots",
                                "format",
                                "duration",
                                "size",
                                "idealScreenSize",
                                "idealScreenDensity",
                                "releaseNotes",
                                "pkgVersion",
                                "semanticVersion",
                                "versionKey",
                                "resources",
                                "downloadUrl",
                                "artifactUrl",
                                "previewUrl",
                                "streamingUrl",
                                "objects",
                                "organization",
                                "createdFor",
                                "developer",
                                "source",
                                "notes",
                                "pageNumber",
                                "publication",
                                "edition",
                                "publisher",
                                "author",
                                "owner",
                                "attributions",
                                "collaborators",
                                "creators",
                                "contributors",
                                "voiceCredits",
                                "soundCredits",
                                "imageCredits",
                                "copyright",
                                "license",
                                "language",
                                "words",
                                "text",
                                "forkable",
                                "translatable",
                                "ageGroup",
                                "interactivityLevel",
                                "contentType",
                                "resourceType",
                                "category",
                                "templateType",
                                "genre",
                                "theme",
                                "themes",
                                "rating",
                                "rating_a",
                                "quality",
                                "genieScore",
                                "authoringScore",
                                "popularity",
                                "downloads",
                                "launchUrl",
                                "activity_class",
                                "draftImage",
                                "scaffolding",
                                "feedback",
                                "feedbackType",
                                "teachingMode",
                                "skills",
                                "keywords",
                                "domain",
                                "dialcodes",
                                "optStatus",
                                "description",
                                "instructions",
                                "body",
                                "oldBody",
                                "stageIcons",
                                "editorState",
                                "data",
                                "loadingMessage",
                                "checksum",
                                "learningObjective",
                                "createdBy",
                                "creator",
                                "reviewer",
                                "lastUpdatedBy",
                                "lastSubmittedBy",
                                "lastSubmittedOn",
                                "lastPublishedBy",
                                "lastPublishedOn",
                                "versionDate",
                                "origin",
                                "originData",
                                "versionCreatedBy",
                                "me_totalSessionsCount",
                                "me_creationSessions",
                                "me_creationTimespent",
                                "me_totalTimespent",
                                "me_totalInteractions",
                                "me_averageInteractionsPerMin",
                                "me_averageSessionsPerDevice",
                                "me_totalDevices",
                                "me_averageTimespentPerSession",
                                "me_averageRating",
                                "me_totalDownloads",
                                "me_totalSideloads",
                                "me_totalRatings",
                                "me_totalComments",
                                "me_totalUsage",
                                "me_totalLiveContentUsage",
                                "me_usageLastWeek",
                                "me_deletionsLastWeek",
                                "me_lastUsedOn",
                                "me_lastRemovedOn",
                                "me_hierarchyLevel",
                                "me_totalDialcodeAttached",
                                "me_totalDialcodeLinkedToContent",
                                "me_totalDialcode",
                                "flagReasons",
                                "flaggedBy",
                                "flags",
                                "lastFlaggedOn",
                                "tempData",
                                "copyType",
                                "pragma",
                                "publishChecklist",
                                "publishComment",
                                "rejectReasons",
                                "rejectComment",
                                "totalQuestions",
                                "totalScore",
                                "ownershipType",
                                "reservedDialcodes",
                                "dialcodeRequired",
                                "lockKey",
                                "badgeAssociations",
                                "framework",
                                "lastStatusChangedOn",
                                "uploadError",
                                "appId",
                                "s3Key",
                                "consumerId",
                                "organisation",
                                "nodeType",
                                "prevState",
                                "publishError",
                                "publish_type",
                                "ownedBy",
                                "purpose",
                                "toc_url",
                                "reviewError",
                                "mimeTypesCount",
                                "contentTypesCount",
                                "childNodes",
                                "leafNodesCount",
                                "depth",
                                "SYS_INTERNAL_LAST_UPDATED_ON",
                                "assets",
                                "version",
                                "qrCodeProcessId",
                                "migratedUrl",
                                "totalCompressedSize",
                                "programId",
                                "leafNodes",
                                "editorVersion",
                                "unitIdentifiers",
                                "questionCategories",
                                "certTemplate",
                                "subject",
                                "medium",
                                "gradeLevel",
                                "topic",
                                "subDomains",
                                "subjectCodes",
                                "difficultyLevel",
                                "board",
                                "licenseterms",
                                "copyrightYear",
                                "organisationId",
                                "itemSetPreviewUrl",
                                "textbook_name",
                                "level1Name",
                                "level1Concept",
                                "level2Name",
                                "level2Concept",
                                "level3Name",
                                "level3Concept",
                                "sourceURL",
                                "me_totalTimeSpentInSec",
                                "me_totalPlaySessionCount",
                                "me_totalRatingsCount",
                                "monitorable",
                                "userConsent",
                                "trackable",
                                "credentials",
                                "processId",
                                "primaryCategory",
                                "additionalCategories",
                                "prevStatus",
                                "cloudStorageKey",
                                "batches",
                                "year",
                                "plugins",
                                "showNotification",
                                "collectionId",
                                "learningOutcome",
                                "displayScore",
                                "boardIds",
                                "gradeLevelIds",
                                "subjectIds",
                                "mediumIds",
                                "topicsIds",
                                "targetFWIds",
                                "targetBoardIds",
                                "targetGradeLevelIds",
                                "targetSubjectIds",
                                "targetMediumIds",
                                "targetTopicIds",
                                "maxAttempts",
                                "se_frameworkIds",
                                "se_boardIds",
                                "se_subjectIds",
                                "se_mediumIds",
                                "se_topicIds",
                                "se_gradeLevelIds",
                                "se_boards",
                                "se_subjects",
                                "se_mediums",
                                "se_topics",
                                "se_gradeLevels"
                            ]
                        }
                    }
                ]
            },
            "created_on": "2024-04-01T08:16:02.020Z",
            "last_modified_on": null,
            "rootOrgId": "*"
        }
    },
    "ts": "2024-04-03T11:49:44.507Z",
    "ver": "1.0"
}`

**Comment by rajeevsathish** [comment]
@siddhishinde0723  in the menu bar the all tab for search is not enabled so please enable that tab and see the solution will work

**Reply by siddhishinde0723** [reply]
Thank you, @rajeevsathish the issue is solved.
